src/csharp/Grpc.IntegrationTesting/ClientRunners.cs

```diff
@@ -55,6 +58,15 @@ public class ClientRunners
     {
         static readonly ILogger Logger = GrpcEnvironment.Logger.ForType<ClientRunners>();
 
+        // Profilers to use for clients.
+        static readonly BlockingCollection<BasicProfiler> profilers = new BlockingCollection<BasicProfiler>();
+
+        internal static void AddProfiler(BasicProfiler profiler)
```

Comment:
But wouldn't you need to know ```outstandingRpcsPerChannel``` beforehand, which is part of the request?  It seems you need to call ClientRunners.AddProfiler() ```outstandingRpcsPerChannel``` times to ensure each thread gets a profiler.